c++/src/kj/test-helpers.c++

```diff
@@ -153,5 +153,72 @@ bool expectFatalThrow(kj::Maybe<Exception::Type> type, kj::Maybe<StringPtr> mess
 #endif
 }
 
+bool expectExit(Maybe<int> statusCode, FunctionParam<void()> code)  noexcept {
+#if _WIN32
+  // We don't support death tests on Windows due to lack of efficient fork.
+  return true;
+#else
+  pid_t child;
+  KJ_SYSCALL(child = fork());
+  if (child == 0) {
+    code();
+    _exit(0);
+  }
+
+  int status;
+  KJ_SYSCALL(waitpid(child, &status, 0));
+
+  if (WIFEXITED(status)) {
+    KJ_IF_MAYBE(s, statusCode) {
+      KJ_EXPECT(WEXITSTATUS(status) == *s);
+      return WEXITSTATUS(status) == *s;
+    } else {
+      KJ_EXPECT(WEXITSTATUS(status) != 0);
+      return WEXITSTATUS(status) != 0;
+    }
+  } else {
+    if (WIFSIGNALED(status)) {
+      KJ_FAIL_EXPECT("subprocess didn't exit but triggered a signal", strsignal(WTERMSIG(status)));
+    }
```

Comment:
`EXPECT` macros do not throw an exception, they just note the failure and keep executing. So you need `} else {` here, or just return false.

Actually on that topic, it's a bit weird that these functions return a boolean in addition to the expectation failures inside. So when there's a failure, actually two failures will be reported. However, this code review has gone on too long already so I'd say don't worry about it, we can clean that up some other time. Plus it looks like `KJ_EXPECT_THROW` has some similar problems.

Reply:
fixed.